### ISSUE TITLE
Automatic conversion of false to array is deprecated

### ISSUE THREAD

**Issue by Chouby** [woothemes/woocommerce#38860]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Running PHPUnit tests with WC 7.9 beta 1 + PHP 8.2, deprecated notices are logged:
```
PHP Deprecated:  Automatic conversion of false to array is deprecated in /home/travis/build/polylang/polylang-wc/tmp/woocommerce/plugins/woocommerce/src/Admin/DataSourcePoller.php on line 138
```

### Expected behavior

No deprecated notice

### Actual behavior

N/A

### Steps to reproduce

N/A (Im' running private PHPUnit tesfs)

### WordPress Environment

WP 6.2
WC 7.9 beta 1
PHP 8.2

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by dominic-p** [comment]
I just observed the same deprecation notice testing my site for PHP 8.1 compatibility.

When I visited `/wp-admin` it said a database upgraded was needed. Once I ran that, the notice went away. I'm not sure if the two events are related, but I thought I would share as it might help with reproduction.

**Comment by rrennick** [comment]
@Chouby Thanks for creating the issue. I just ran the tests in PHP 8.2 and don't see this notice. Can you let us know whether you are still seeing it?

**Comment by Chouby** [comment]
Yes I do. Reading [the code 2 lines above the notice](https://github.com/woocommerce/woocommerce/blob/8.2.0-beta.1/plugins/woocommerce/src/Admin/DataSourcePoller.php#L136), `get_transient()` can return false. 2 lines later the code is wrongly expecting that `$specs_group` is always an array.

As a side note, using a static analysis tool such as PHPStan would help you to avoid this kind of bug.

**Comment by psealock** [comment]
> Yes I do. Reading [the code 2 lines above the notice](https://github.com/woocommerce/woocommerce/blob/8.2.0-beta.1/plugins/woocommerce/src/Admin/DataSourcePoller.php#L136), get_transient() can return false. 2 lines later the code is wrongly expecting that $specs_group is always an array.

Looks like thats the case to me. Something like this will be the one I think:

```php
$transient   = get_transient( $this->args['transient_name'] );
$specs_group = $transient ? $transient : array();
```
@Chouby Would you like to open a pull request?

**Comment by dominic-p** [comment]
I don't think this is stale. It looks like it's just waiting for a PR at the moment. I just saw these notices at the bottom of the WC Status admin page after updating to version 8.5.2:

```
Deprecated: Automatic conversion of false to array is deprecated in /srv/www/public_html/wp-content/plugins/woocommerce/src/Admin/DataSourcePoller.php on line 138

Deprecated: Automatic conversion of false to array is deprecated in /srv/www/public_html/wp-content/plugins/woocommerce/src/Admin/DataSourcePoller.php on line 138
```

**Comment by chihsuan** [comment]
This has been fixed in https://github.com/woocommerce/woocommerce/pull/44384 and will be released in WC 8.7.0
